crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -311,63 +316,64 @@ where
             .collect()
     }
 
-    /// Read a lock for a specific (transaction, shared object) pair.
-    pub fn get_all_shared_locks(
+    /// Read shared object locks / versions for a specific transaction.
+    pub fn get_shared_locks(
         &self,
         transaction_digest: &TransactionDigest,
     ) -> Result<Vec<(ObjectID, SequenceNumber)>, SuiError> {
         Ok(self
             .tables
-            .assigned_object_versions
-            .iter()
-            .skip_to(&(*transaction_digest, ObjectID::ZERO))?
-            .take_while(|((tx, _objid), _ver)| tx == transaction_digest)
-            .map(|((_tx, objid), ver)| (objid, ver))
-            .collect())
+            .assigned_shared_object_versions
+            .get(transaction_digest)?
+            .unwrap_or_default())
     }
 
     #[cfg(test)]
     pub fn get_next_object_version(&self, obj: &ObjectID) -> Option<SequenceNumber> {
-        self.tables.next_object_versions.get(obj).unwrap()
+        self.tables.next_shared_object_versions.get(obj).unwrap()
     }
 
     /// Read a lock for a specific (transaction, shared object) pair.
     #[cfg(test)] // Nothing wrong with this function, but it is not currently used outside of tests
-    pub fn get_assigned_object_versions<'a>(
+    pub fn get_assigned_shared_object_versions<'a>(
```

Comment:
it seems like we could get rid of this method and only use `get_shared_locks` instead. Maybe leave a TODO?